### PR TITLE
[Benchmark] Add TPC-H benchmark tools

### PR DESCRIPTION
**Pull request by hello-stephen**
# Proposed changes

Issue Number: empty

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No
3. Has document been added or modified: No
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: tools/tpch-tools/run-tpch-queries.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+pre_set() {
+    echo $@
+    mysql -h$FE_HOST -u$USER -P$FE_QUERY_PORT -D$DB -e "$@"
+}
+
+pre_set "set global enable_vectorized_engine=1;"
+pre_set "set global parallel_fragment_exec_instance_num=8;"
+pre_set "set global exec_mem_limit=48G;"
+pre_set "set global batch_size=1024;"
```
Comment (BiteTheDDDDt) [inline tools/tpch-tools/run-tpch-queries.sh:104]: here can set batch_size=4096

**Review by hf200012 — DISMISSED** [review]
LGTM

File: tools/tpch-tools/build-tpch-dbgen.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+check_prerequest "unzip -h" "unzip"
+
+# download tpch tools pacage first
+if [[ -d $TPCH_DBGEN_DIR ]]; then
+    echo "Dir $TPCH_DBGEN_DIR already exists. No need to download."
+    echo "If you want to download TPC-H_Tools_v3.0.0 again, please delete this dir first."
+else
+    # download from internal
+    wget "https://tools-chengdu.oss-cn-chengdu-internal.aliyuncs.com/TPC-H_Tools_v3.0.0.zip"
```
Comment (morningman) [inline tools/tpch-tools/build-tpch-dbgen.sh:54]: internal?

File: tools/tpch-tools/run-tpch-queries.sh (hunk 1 above)
Comment (hello-stephen) [reply tools/tpch-tools/run-tpch-queries.sh]: ok, set to 4096

File: tools/tpch-tools/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+## Usage
```
[the remark this answers is not in the document]
Comment (hello-stephen) [reply tools/tpch-tools/README.md:20]: added

File: tools/tpch-tools/build-tpch-dbgen.sh (hunk 2 above)
Comment (hello-stephen) [reply tools/tpch-tools/build-tpch-dbgen.sh]: I have removed the internal url.

File: tools/tpch-tools/run-tpch-queries.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+echo "USER: $USER"
+echo "PASSWORD: $PASSWORD"
+echo "DB: $DB"
+
+pre_set() {
+    echo $@
+    mysql -h$FE_HOST -u$USER -P$FE_QUERY_PORT -D$DB -e "$@"
+}
+
+pre_set "set global enable_vectorized_engine=1;"
```
Comment (zbtzbtzbt) [inline tools/tpch-tools/run-tpch-queries.sh:101]: thanks your pr！

I have some questions
1.Can we specify the size of the generated data？
2.Can we specify the params？for example `enable_vectorized_engine=false`

following is my opinion：
1.we can user add more params such as `runtime filter、join reorder and so on`
2.can we simplify the steps？i think five steps are too complicated，can we merge step3 and step4？

Comment (BiteTheDDDDt) [reply tools/tpch-tools/run-tpch-queries.sh:101]: > thanks your pr！
> 
> I have some questions 1.Can we specify the size of the generated data？ 2.Can we specify the params？for example `enable_vectorized_engine=false`
> 
> following is my opinion： 1.we can user add more params such as `runtime filter、join reorder and so on` 2.can we simplify the steps？i think five steps are too complicated，can we merge step3 and step4？

Further, maybe we can unify a benchmark framework and let it support test sets such as ssb/tpch/tpcds...

File: tools/tpch-tools/load-tpch-data.sh [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+# start load
+load_region $TPCH_DATA_DIR/region.tbl
+load_nation $TPCH_DATA_DIR/nation.tbl
+load_supplier $TPCH_DATA_DIR/supplier.tbl
+load_customer $TPCH_DATA_DIR/customer.tbl
+load_part $TPCH_DATA_DIR/part.tbl
+
+# set parallelism
+
+# 以PID为名, 防止创建命名管道时与已有文件重名，从而失败
```
Comment (zbtzbtzbt) [inline tools/tpch-tools/load-tpch-data.sh:175]: use English

File: tools/tpch-tools/run-tpch-queries.sh (hunk 4 above)
Comment (zbtzbtzbt) [reply tools/tpch-tools/run-tpch-queries.sh:101]: > Further, maybe we can unify a benchmark framework and let it support test sets such as ssb/tpch/tpcds...

good proposal

File: tools/tpch-tools/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+### 1. build tpc-h dbgen tool.
+    ./build-tpch-dbgen.sh
+### 2. generate tpc-h data. use -h for more infomations.
+    ./gen-tpch-data.sh -s 1
+### 3. create tpc-h tables. modify `doris-cluster.conf` to specify doris info, then run script below.
+    ./create-tpch-tables.sh
+### 4. load tpc-h data. use -h for help.
+    ./load-tpch-data.sh
+### 5. run tpc-h queries.
+    ./run-tpch-queries.sh
```
Comment (EmmyMiao87) [inline tools/tpch-tools/README.md:34]: In fact, the test set query of tpch is slightly different under different data volumes.
So it's best to explain in the document that the query we give is a query under how big a dataset.

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**